OsmAnd/src/net/osmand/plus/plugins/odb/dialogs/OBDMainFragment.kt

```diff
@@ -301,11 +321,17 @@ class OBDMainFragment : OBDDevicesBaseFragment(), VehicleMetricsPlugin.Connectio
 
 	override fun onNameChanged() {
 		view?.findViewById<TextView>(R.id.device_name)?.text = device.name
-		adapter.notifyDataSetChanged()
+		adapter?.notifyDataSetChanged()
 	}
 
 	override fun onForgetSensorConfirmed(deviceId: String) {
 		vehicleMetricsPlugin?.removeDeviceToUsedOBDDevicesList(deviceId)
 		view?.let { setupUI(it) }
 	}
+
+	override fun updateNightMode() {
+		super.updateNightMode()
+		adapter?.nightMode = nightMode
+		adapter?.notifyDataSetChanged()
```

Comment:
It looks somewhat strange, correct nightMode should be set in onCreateView, also theme change should force layout recreation after this [method](https://github.com/osmandapp/Osmand/blob/fecbd55b6d6cd4b30a232b8ab0e1b2ba2bceadf8/OsmAnd/src/net/osmand/plus/helpers/MapFragmentsHelper.java#L117)